crates/fluvio-cli/src/smartmodule/generate.rs

```diff
@@ -0,0 +1,115 @@
+use std::path::PathBuf;
+
+use tracing::info;
+use clap::{Parser, ArgEnum};
+use duct::{cmd};
+use include_dir::{Dir, include_dir};
+
+use fluvio::Fluvio;
+use fluvio_cli_common::install::fluvio_base_dir;
+
+use crate::Result;
+use super::fluvio_smart_dir;
+
+const TEMPLATE_DIR: Dir = include_dir!("$CARGO_MANIFEST_DIR/templates");
+
+/// return template directory for each CLI version
+/// if it doesn't exists, create template directory and unpack from inline template
+fn fluvio_template_dir() -> Result<PathBuf> {
+    let base_dir = fluvio_base_dir()?;
+    let template_path = base_dir.join("templates").join(env!("GIT_HASH"));
+    if !template_path.exists() {
+        info!(log = ?template_path,"creating smart module template directory");
+        std::fs::create_dir_all(&template_path)?;
+        TEMPLATE_DIR.extract(&template_path)?;
+    }
+    Ok(template_path)
+}
+
+#[derive(ArgEnum, Debug, Clone, PartialEq)]
+#[allow(non_camel_case_types)]
+pub enum DataType {
+    json,
+}
+
+impl Default for DataType {
```

Comment:
Starting with [rust 1.62](https://blog.rust-lang.org/2022/06/30/Rust-1.62.0.html) you can use derive default on enum variants. A bit nicer to read IMHO.
 ```rust
#[derive(ArgEnum, Default, Debug, Clone, PartialEq)]
#[allow(non_camel_case_types)]
pub enum DataType {
    #[default]
    json,
}
```